guides/keras_cv/fine_tune_via_textual_inversion.py

```diff
@@ -0,0 +1,691 @@
+new_vocab_size = len(stable_diffusion.tokenizer.vocab)
+
+# SAme hack here to get embedding layer from layers[2]
+old_token_weights = stable_diffusion.text_encoder.layers[
+    2
+].token_embedding.get_weights()
+old_position_weights = stable_diffusion.text_encoder.layers[
+    2
+].position_embedding.get_weights()
+
+old_token_weights = old_token_weights[0]
+new_weights = np.expand_dims(new_weights, axis=0)
+new_weights = np.concatenate([old_token_weights, new_weights], axis=0)
+
+
+"""
+Lets construct a new TextEncoder and prepare it.
+"""
+
+# Have to set download_weights False so we can init (otherwise tries to load weights)
+new_encoder = keras_cv.models.stable_diffusion.TextEncoder(
+    keras_cv.models.stable_diffusion.stable_diffusion.MAX_PROMPT_LENGTH,
+    vocab_size=new_vocab_size,
+    download_weights=False,
+)
+for index, layer in enumerate(stable_diffusion.text_encoder.layers):
+    # Layer 2 is the embedding layer, so we omit it from our weight-copying
+    if index == 2:
+        continue
+    new_encoder.layers[index].set_weights(layer.get_weights())
+
+
+new_encoder.layers[2].token_embedding.set_weights([new_weights])
+new_encoder.layers[2].position_embedding.set_weights(old_position_weights)
+
+stable_diffusion._text_encoder = new_encoder
+stable_diffusion._text_encoder.compile(jit_compile=True)
+
+"""
+## Training
+
+Now we can move on to the exciting part: training!
+
+In TextualInversion the only piece of the model that is trained is the embedding vector.
+Lets freeze the rest of the model.
+"""
+
+
+stable_diffusion.diffusion_model.trainable = False
+stable_diffusion.decoder.trainable = False
+stable_diffusion.text_encoder.trainable = True
+
+stable_diffusion.text_encoder.layers[2].trainable = True
+
+
+def traverse_layers(layer):
+    if hasattr(layer, "layers"):
+        for layer in layer.layers:
+            yield layer
+    if hasattr(layer, "token_embedding"):
+        yield layer.token_embedding
+    if hasattr(layer, "position_embedding"):
+        yield layer.position_embedding
+
+
+for layer in traverse_layers(stable_diffusion.text_encoder):
+    if isinstance(layer, keras.layers.Embedding) or "clip_embedding" in layer.name:
+        layer.trainable = True
+    else:
+        layer.trainable = False
+
+new_encoder.layers[2].position_embedding.trainable = False
+
+"""
+Lets confirm the proper weights are set to trainable.
+"""
+all_models = [
+    stable_diffusion.text_encoder,
+    stable_diffusion.diffusion_model,
+    stable_diffusion.decoder,
+]
+print([[w.shape for w in model.trainable_weights] for model in all_models])
+
+"""
+## Training the new embedding
+In order to train the embedding, we'll need a couple of utilities.
+We import a NoiseScheduler from KerasCV, and define the following utilities below:
+
+- `_sample_from_encoder_outputs` is a wrapper around the base StableDiffusion text
+encoder which samples from the statistical distribution produced by the SG image
```

Comment:
probably just drop "SG" here (which was supposed to be "SD")

(Same on the line below)